Lesson1/Dima_Gnatyk/main.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <vector>
+#include <list>
+#include <forward_list>
+#include <numeric>
+
+template <class T>
+void print(const std::vector<T> &data){
+    for(auto value:data){
+        std::cout <<value<<" ";
+    }
+    std::cout<<std::endl;
+}
+
+template <class T>
+std::vector<T> concat(const std::vector<T> &data_vector_1, const std::vector<T> &data_vector_2){
+    if(data_vector_1.empty()) return data_vector_2;
+    if(data_vector_2.empty()) return data_vector_1;
+
+    std::vector<T> union_vector {};
+    union_vector.reserve(data_vector_1.size()+data_vector_2.size());
+    union_vector.insert(union_vector.cend(), data_vector_1.cbegin(), data_vector_1.cend());
+    union_vector.insert(union_vector.cend(), data_vector_2.begin(), data_vector_2.end());
+
+    return union_vector;
+}
+
+template <class T>
+std::list<T> new_number(const T &number, const T &pow){
+    std::list<T> result {};
+   // if(number > 0) return result.push_back(0);
```

Comment:
Always do not forget to remove the commented lines when you are pushing your changes.
Changes are not lost, you always can find it with git log.